specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/common/2023-05-01-preview/definitions.json

```diff
@@ -0,0 +1,168 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2023-05-01-preview",
+    "title": "Common types and definitions",
+    "description": "Definitions and parameters common for all resource types of KubernetesConfiguration RP"
+  },
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {},
+  "definitions": {
+    "ProvisioningStateDefinition": {
+      "type": "string",
+      "readOnly": true,
+      "description": "The provisioning state of the resource.",
+      "enum": [
+        "Succeeded",
+        "Failed",
+        "Canceled",
+        "Creating",
+        "Updating",
+        "Deleting"
+      ],
+      "x-ms-enum": {
+        "name": "ProvisioningState",
+        "modelAsString": true
+      }
+    },
+    "OperationStatusResult": {
```

Comment:
>OperationStatusResult [](http://example.com/codeflow?start=5&length=21)

Instead of defining your own OperationStatusResult, can you use the common type definition?

Reply:
I'm not really sure where that came from or why...I don't see it being used anywhere...